metastore/src/java/org/apache/hadoop/hive/metastore/ReplChangeManager.java

```diff
@@ -133,11 +133,13 @@ void addFile(Path path) throws MetaException {
    * Move a path into cmroot. If the path is a directory (of a partition, or table if nonpartitioned),
    *   recursively move files inside directory to cmroot. Note the table must be managed table
    * @param path a single file or directory
-   * @param ifPurge if the file should skip Trash when delete
+   * @param isMove if the files to be copied or moved to cmpath. Copy if false and move if true.
+   * @param ifPurge if the file should skip Trash when delete.
+   *                This is referred only if isMove is true.
    * @return int
    * @throws MetaException
    */
-  int recycle(Path path, boolean ifPurge) throws MetaException {
+  int recycle(Path path, boolean isMove, boolean ifPurge) throws MetaException {
```

Comment:
I think boolean representation of the operation type copy/move is confusing, is it possible to use an enum type such that operation type can be provided, This will explicitly provide information wherever the method is called as to what operation is to be performed rather than trying to figure out the boolean mapping for the operation type ?